src/backends/interflop-cancellation/interflop_cancellation.c

```diff
@@ -67,28 +67,50 @@ static void _set_tolerance(int tolerance) { TOLERANCE = tolerance; }
 static void _set_warning(bool warning) { WARN = warning; }
 
 /* random generator internal state */
-static tinymt64_t random_state;
+// static tinymt64_t random_state;
 
-/* Set the mca seed */
-static void _set_mca_seed(const bool choose_seed, const uint64_t seed) {
-  _set_seed_default(&random_state, choose_seed, seed);
-}
+/* random number generator internal state */
+static __thread struct drand48_data random_state;
+/* random number generator initialization flag */
+static __thread bool random_state_valid = false;
+
+/* global thread id access lock */
+static pthread_mutex_t global_tid_lock = PTHREAD_MUTEX_INITIALIZER;
+/* global thread identifier */
+static unsigned long long int global_tid = 0;
 
-static double _mca_rand(void) {
-  /* Returns a random double in the (0,1) open interval */
-  return tinymt64_generate_doubleOO(&random_state);
+/* helper data structure to centralize the data used for random number
+ * generation */
+static __thread mca_data_t *mca_data;
```

Comment:
mca_data structure seems to contain pointers to many __thread local variables. There seems to be a bit of overlap between mca_data structure and the tls variables ?

Could we instead keep only the `mca_data` structure; instead of having the duplication ?